### PR TITLE
Chore: add chromatic deploy, precommit message checks, codeql analysis action

### PR DESCRIPTION
**Pull request by ctran4347**
Added several github actions (chromatic and codeql) as seen below in the checks and deploys. Added husky hook for commitizen so commits following repo's commit convention. Feel free to try it out when you run `git commit`

### CODE REVIEW

**Review by herman-rogers — APPROVED** [review]

File: .husky/prepare-commit-msg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+exec < /dev/tty && node_modules/.bin/cz --hook || true
```
Comment (knakamura13) [inline .husky/prepare-commit-msg:2]: I don't understand what this does 😬 

I get what `cz` is but what does this whole command do in unison?

Comment (knakamura13) [reply .husky/prepare-commit-msg:2]: Doesn't commitizen already prevent commits from straying from conventions?

**Comment by knakamura13** [comment]
Looks good to me other than the Husky change that I don't fully understand (is it redundant? no idea).

**Review by knakamura13 — COMMENTED** [review]
Looks good to me other than the Husky change that I don't fully understand (is it redundant? no idea).

File: .husky/prepare-commit-msg (hunk 1 above)
Comment (ctran4347) [reply .husky/prepare-commit-msg:2]: `exec < /dev/tty` opens a filesystem object representing the terminal process. If that succeeds then, ` node_modules/.bin/cz --hook` runs the commitizen. `|| true` checks if the entire `exec < /dev/tty && node_modules/.bin/cz --hook` execution succeed, if not it'll output `true`.

Commitizen does prevent commits from straying from convention after the commit message has been submitted such as `git commit -m "commitizen sucks"`. However, the husky/git hook that I added will force the user whenever they run `git commit` or `git commit -m "some messager"` to fill out a commitizen template in the terminal.

The `prepare-commit-msg` hook is for enforcing commitizen and the `pre-commit` hook is for checking the commit meets commitizen/commit conventions

**Review by knakamura13 — APPROVED** [review]

**Review by disposedtrolley — APPROVED** [review]
Looks great! I’m particularly excited for the CodeQL scanning, I haven’t had a chance to see it in action yet.

**Comment by disposedtrolley** [comment]
Bizarre. I rebased this on dev as the branch rules require PR branches to be up-to-date with their targets, but this has caused the chromatic builds to fail.

**Comment by ctran4347** [comment]
> Bizarre. I rebased this on dev as the branch rules require PR branches to be up-to-date with their targets, but this has caused the chromatic builds to fail.

There's currently a change in dev that is causing storybook to error out. We are sorting that out in the slack/discord!

**Pull request merged**